Dockerfile

```diff
@@ -1,11 +1,17 @@
 FROM ruby:3.2.2
 
 RUN apt-get update -qq && \
-    apt-get install -y build-essential libssl-dev nodejs libpq-dev less vim nano libsasl2-dev
-
+    apt-get install -y build-essential libssl-dev libpq-dev less vim nano libsasl2-dev
 RUN curl -sS https://dl.yarnpkg.com/debian/pubkey.gpg | apt-key add -
 RUN echo "deb https://dl.yarnpkg.com/debian/ stable main" | tee /etc/apt/sources.list.d/yarn.list
-RUN apt update && apt install yarn
+
+RUN mkdir -p /etc/apt/keyrings
+RUN curl -fsSL https://deb.nodesource.com/gpgkey/nodesource-repo.gpg.key | gpg --dearmor -o /etc/apt/keyrings/nodesource.gpg
+
+ENV NODE_MAJOR=18
+RUN echo "deb [signed-by=/etc/apt/keyrings/nodesource.gpg] https://deb.nodesource.com/node_$NODE_MAJOR.x nodistro main" | tee /etc/apt/sources.list.d/nodesource.list
+
+RUN apt update && apt install -y yarn nodejs
```

Comment:
We can also install node using binaries https://github.com/nodejs/help/wiki/Installation#how-to-install-nodejs-via-binary-archive-on-linux